server/src/main/java/io/unitycatalog/server/persist/utils/PagedEntityLister.java

```diff
@@ -0,0 +1,90 @@
+package io.unitycatalog.server.persist.utils;
+
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.persist.dao.*;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+public class PagedEntityLister<T extends IdentifiableDAO> {
```

Comment:
shall we call it `PagedListingHelper`?

please also add javadoc for the class and methods